src/pykeen/utils.py

```diff
@@ -917,6 +930,43 @@ def extend_batch(
     return hrt_batch
 
 
+def check_shapes(
+    *x: Tuple[Union[torch.Tensor, Tuple[int, ...]], str],
+    raise_on_errors: bool = True,
+) -> bool:
+    """Verify that a sequence of tensors are of matching shapes.
+
+    :param x:
```

Comment:
from the type annotation / code below it looks like the first element of the tuple can also be a tensor's shape, i.e. a tuple of integers.